worlds/kh1/__init__.py

```diff
@@ -0,0 +1,315 @@
+                    ,"required_reports_door": int(required_reports_door)
+                    ,"door": self.options.final_rest_door.current_key
+                    ,"seed": self.multiworld.seed_name
+                    ,"advanced_logic": bool(self.options.advanced_logic)
+                    ,"hundred_acre_wood": bool(self.options.hundred_acre_wood)
+                    ,"atlantica": bool(self.options.atlantica)
+                    ,"goal": str(self.options.goal.current_key)}
+        if self.options.randomize_keyblade_stats:
+            min_str_bonus = min(self.options.keyblade_min_str, self.options.keyblade_max_str)
+            max_str_bonus = max(self.options.keyblade_min_str, self.options.keyblade_max_str)
+            min_mp_bonus = min(self.options.keyblade_min_mp, self.options.keyblade_max_mp)
+            max_mp_bonus = max(self.options.keyblade_min_mp, self.options.keyblade_max_mp)
+            slot_data["keyblade_stats"] = ""
+            for i in range(22):
+                if i < 4 and self.options.bad_starting_weapons:
+                    slot_data["keyblade_stats"] = slot_data["keyblade_stats"] + "1,0,"
+                else:
+                    if min_str_bonus != max_str_bonus:
+                        str_bonus = int(self.random.randrange(min_str_bonus,max_str_bonus))
+                    else:
+                        str_bonus = int(min_str_bonus)
+                    if min_mp_bonus != max_mp_bonus:
+                        mp_bonus = int(self.random.randrange(min_mp_bonus,max_mp_bonus))
+                    else:
+                        mp_bonus = int(min_mp_bonus)
+                    slot_data["keyblade_stats"] = slot_data["keyblade_stats"] + str(str_bonus) + "," + str(mp_bonus) + ","
+            slot_data["keyblade_stats"] = slot_data["keyblade_stats"][:-1]
+        if self.options.donald_death_link:
+            slot_data["donalddl"] = ""
+        if self.options.goofy_death_link:
+            slot_data["goofydl"] = ""
+        if self.options.keyblades_unlock_chests:
+            slot_data["chestslocked"] = ""
+        else:
+            slot_data["chestsunlocked"] = ""
+        if self.options.interact_in_battle:
+            slot_data["interactinbattle"] = ""
+        return slot_data
+    
+    def create_item(self, name: str) -> KH1Item:
+        data = item_table[name]
+        return KH1Item(name, data.classification, data.code, self.player)
+
+    def create_event(self, name: str) -> KH1Item:
+        data = event_item_table[name]
+        return KH1Item(name, data.classification, data.code, self.player)
+
+    def set_rules(self):
+        set_rules(self.multiworld, self.player, self.options, self.determine_reports_required_to_open_end_of_the_world(), self.determine_reports_required_to_open_final_rest_door(), self.options.final_rest_door.current_key)
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, self.options)
```

Comment:
I would recommend reworking these to just pass in world so that it's just
`set_rules(self)`
and then you can just get all of these properties off of the world object. But, it's up to you really

Reply:
Reworked to use self then pull needed values out inside the set_rules function in Rules.py